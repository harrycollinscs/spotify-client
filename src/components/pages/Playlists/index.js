import { connect } from 'react-redux';
import Playlists from '../../templates/Playlists';

const mapStateToProps = () => {

}

const mapDispatchToProps = () => {
  
}

export default connect()(Playlists);
