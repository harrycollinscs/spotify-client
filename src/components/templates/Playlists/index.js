import axios from "axios";
import React, { PureComponent } from "react";
import styled from "styled-components";
import { me_endpoint } from "../../../config";
import store from "../../../store";
import CTA from "../../atoms/CTA";
import Page from "../../atoms/Page";
import Hero from "../../molecules/Hero";
import PlaylistList from "../../organisms/PlaylistList";

const ContentContainer = styled.div`
  padding: 30px 100px 30px 100px;
`;

const PlaylistsContainer = styled.div`
  display: flex;
  flex-direction: column;
  border-radius: 5px;
  background-color: black;
  align-items: center;
  padding: 20px 0px 50px 0px;
  text-align: center;
`;

const StyledTitle = styled.h2`
  font-weight: bold;
  padding-bottom: 20px;
`;

class Playlists extends PureComponent {
  state = {
    loadedPlaylists: [],
    totalPlaylists: null,
  };

  componentDidMount() {
    this.loadPlaylists(this.state.loadedPlaylists.length);
  }

  loadPlaylists(offset) {
    const playlistsEndpoint = `${me_endpoint}/playlists`;

    axios
      .get(playlistsEndpoint, {
        headers: {
          Authorization: "Bearer " + store.getState().accessToken,
        },
        params: {
          limit: 30,
          offset: offset,
        },
      })
      .then((res) => {
        this.setState({
          loadedPlaylists: [...this.state.loadedPlaylists, ...res.data.items],
          totalPlaylists: res.data.total,
        });
      });
  }

  locale = {
    heroTitle: "Playlists",
    heroContent: "Here you can find the playlists you listen to most often",
  };

  render() {
    const playlists = this.state.loadedPlaylists;
    const allPlaylistsLoaded = this.state.totalPlaylists === playlists.length;

    return (
      <Page>
        <Hero title={this.locale.heroTitle} content={this.locale.heroContent} />
        <ContentContainer>
          <StyledTitle>Your playlists</StyledTitle>
          {playlists.length ? (
            <PlaylistsContainer>
              <PlaylistList playlists={playlists} />
              {!allPlaylistsLoaded && (
                <CTA
                  action={() =>
                    this.loadPlaylists(this.state.loadedPlaylists.length)
                  }
                  primary={false}
                >
                  Load more
                </CTA>
              )}
            </PlaylistsContainer>
          ) : (
            <p>No data available</p>
          )}
        </ContentContainer>
      </Page>
    );
  }
}

export default Playlists;
