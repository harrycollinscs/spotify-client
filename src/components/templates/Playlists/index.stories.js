import React from "react";
import Playlists from "./index";

export default {
  title: "Templates/Playlists",
  component: Playlists,
};

const Template = (args) => <Playlists {...args} />;

export const Default = Template.bind({});
