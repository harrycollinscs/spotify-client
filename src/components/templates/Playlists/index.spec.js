import React from "react";
import Playlists from "./index";
import { shallow } from "enzyme";

describe("Templates/Playlists", () => {
  const component = <Playlists />;

  it("Should match snapshot", () => {
    const render = shallow(component);
    expect(render).toMatchSnapshot();
  });
});
