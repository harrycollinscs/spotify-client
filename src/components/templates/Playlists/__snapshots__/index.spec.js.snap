// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/Playlists Should match snapshot 1`] = `
<Page>
  <Hero
    content="Here you can find the playlists you listen to most often"
    title="Playlists"
  />
  <styled.div>
    <styled.h2>
      Your playlists
    </styled.h2>
    <styled.div>
      0
      <Loader
        color="white"
        css={true}
        loading={true}
        size={40}
      />
    </styled.div>
  </styled.div>
</Page>
`;
